src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -1426,23 +1426,51 @@ class LedgerMasterImp
         if (!referenceLedger || (referenceLedger->getLedgerSeq() < index))
             return ledgerHash; // Nothing we can do. No validated ledger.
 
-        // See if the hash for the ledger we need is in the reference ledger
+        // Is the hash for the ledger we need in the reference ledger?
         ledgerHash = referenceLedger->getLedgerHash (index);
-        if (ledgerHash.isZero())
+
+        // Yes
+        if (ledgerHash.isNonZero())
+            return ledgerHash;
+
+        // No. We need to get another ledger that might have it. That ledger
+        // must meet two requirements:
+        //  1) It must have the hash of the ledger we are looking for.
+        //      This means that its sequence must be equal to greater than the
+        //      sequence that we want but not more than 256 greater.
+        //  2) Its hash must be easy for us to find.
+        //      This means it must be 0 mod 256, because every such ledger is
+        //      permanently enshrined in a LedgerHashes page.
+
+        LedgerIndex const refIndex = (index + 255) & (~255);
+        LedgerHash const refHash = referenceLedger->getLedgerHash (refIndex);
+
+        bool const nonzero (refHash.isNonZero ());
```

Comment:
Quibble:  you're using the `=` form of initialization everywhere else in the change list but here...